pytensor/tensor/rewriting/basic.py

```diff
@@ -941,6 +941,11 @@ def local_sum_make_vector(fgraph, node):
     elements = array.owner.inputs
     acc_dtype = node.op.acc_dtype
     out_dtype = node.op.dtype
+
+    # Skip rewrite when acc_dtype has higher precision than out_dtype
+    if is_an_upcast(out_dtype, acc_dtype):
```

Comment:
This seems less restrictive and still avoid the original issue?

```suggestion
    if acc_dtype == "float64" and pytensor.config.floatX == "float32":
```

Reply:
The issue could also appear if the user requests even less precision. For example, if `config.floatX = "float16"`, then `acc_dtype` will be `float32` and we'll have the same problem. The issue will always exist when `acc_dtype` is more precise than `out_dtype`, which is true for every `floatX` except `float64`.

Reply:
I take your point though. I'll re-work this.

Reply:
Do we support floatX=16 btw? I think float16 was only ever an experimental feature in the codebase and not widely supported